modules/dxp/apps/search-experiences/search-experiences-web/test/js/shared/CodeMirrorEditor.js

```diff
@@ -0,0 +1,312 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {render} from '@testing-library/react';
+import CodeMirror from 'codemirror';
+import React from 'react';
+
+import '@testing-library/jest-dom/extend-expect';
+
+import {getCodeMirrorHints} from '../../../src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/CodeMirrorEditor';
+import aggregationConfigurationSchema from '../../../src/main/resources/META-INF/resources/sxp_blueprint_admin/schemas/aggregation-configuration.schema.json';
+import parameterConfigurationSchema from '../../../src/main/resources/META-INF/resources/sxp_blueprint_admin/schemas/parameter-configuration.schema.json';
+import sortConfigurationSchema from '../../../src/main/resources/META-INF/resources/sxp_blueprint_admin/schemas/sort-configuration.schema.json';
+import sxpQueryElementSchema from '../../../src/main/resources/META-INF/resources/sxp_blueprint_admin/schemas/sxp-query-element.schema.json';
+
+const availableLanguages = {
+	de_DE: 'German (Germany)',
+	en_US: 'English (United States)',
+	es_ES: 'Spanish (Spain)',
+	fr_FR: 'French (France)',
+};
+
+global.document.body.createTextRange = () => {
+	return {
+		getBoundingClientRect: () => {},
+		getClientRects: () => [],
+	};
+};
+
+function createCodeMirrorEditor(props) {
+	document.createElement('div');
+
+	const codeMirror = CodeMirror(document.body, {
+		hintOptions: {
+			completeSingle: false,
+		},
+		mode: {globalVars: true, name: 'application/json'},
+		value: '',
+		...props,
+	});
+
+	codeMirror.execCommand('goLineEnd');
+
+	return codeMirror;
+}
+
+describe('CodeMirrorEditor', () => {
+	it('gets the correct hints', () => {
```

Comment:
Can we make this more specific so it's more helpful what's being tested like `gets the hints for the root level of an object`?

Reply:
Ok! I used that phrase but tweaked a word or two 😅